indexer/src/view/balance.ts

```diff
@@ -0,0 +1,65 @@
+import { Index, PrimaryColumn, ViewColumn, ViewEntity } from 'typeorm';
+
+@ViewEntity({
+  expression: `
+      with transfers as (
+          select (content -> 'to')::numeric "to",
+                 (content -> 'from_')::numeric "from",
+                 (content -> 'value')::numeric "value",
+                 "createdAt" "createdAt"
+          from event
+          where name='Transfer'
+      ),
+      incoming as (
+          select "to",
+                 sum("value") as incoming_credits,
+                 MIN("createdAt") as oldestTransaction,
+                 MAX("createdAt") as newestTransaction
+          from transfers
+          group by "to"
+      ), 
+      outgoing as (
+          select "from",
+          sum("value") as outgoing_credits,
+          MIN("createdAt") as oldestTransaction,
+          MAX("createdAt") as newestTransaction
+      from transfers
+      group by "from"
+      ) select "to" as "userId",
+               (incoming_credits - outgoing_credits) as balance,
+               (
+                  case
+                      when incoming.oldesttransaction>outgoing.oldesttransaction then outgoing.oldesttransaction
+                      else incoming.oldesttransaction
+                      end
+                ) as "createdAt",
+                (
+                  case
+                      when incoming.newesttransaction<outgoing.newesttransaction then outgoing.newesttransaction
+                      else incoming.newesttransaction
+                      end
+                ) as "updatedAt"
+      from incoming 
+          left join outgoing
+                    on "to" = "from"
+                    where "to" != '0';
```

Comment:
We're missing minting and burn events here.

Reply:
This is just filtering out the "0" user from resulting table; I figured it is not necessery to show it here, since we are just interested in users' balances.

Reply:
Yup, and both operations are included in the view (I sum them in `incoming` and `outgoing` per user, and then substract one from the other; in `where` statement here I just exclude user 0 from the resulting summary, since this is probably not needed to anything)